tests/trustless_gateway_car_test.go

```diff
@@ -387,10 +387,80 @@ func TestTrustlessCarDagScopeAll(t *testing.T) {
 }
 
 func TestTrustlessCarEntityBytes(t *testing.T) {
-	singleLayerHamtMultiBlockFilesFixture := car.MustOpenUnixfsCar("t0118/single-layer-hamt-with-multi-block-files.car")
-	subdirWithMixedBlockFiles := car.MustOpenUnixfsCar("t0118/subdir-with-mixed-block-files.car")
+	singleLayerHamtMultiBlockFilesFixture := car.MustOpenUnixfsCar("trustless_gateway_car/single-layer-hamt-with-multi-block-files.car")
+	subdirWithMixedBlockFiles := car.MustOpenUnixfsCar("trustless_gateway_car/subdir-with-mixed-block-files.car")
+	missingBlockFixture := car.MustOpenUnixfsCar("trustless_gateway_car/file-3k-and-3-blocks-missing-block.car")
 
 	tests := SugarTests{
+		{
+			Name: "GET CAR with entity-bytes succeeds even if the gateway is missing some blocks",
+			Hint: `
+				dag-scope=entity&entity-bytes=0:x should return a CAR file with only the blocks needed.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", missingBlockFixture.MustGetCidWithCodec(0x70)).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "0:1000"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						HasRoot(missingBlockFixture.MustGetCid()).
+						MightHaveNoRoots().
+						HasBlocks(
+							missingBlockFixture.MustGetCid(),
+							missingBlockFixture.MustGetChildrenCids()[0],
+						).
+						Exactly(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes succeeds even if the gateway is missing some blocks",
+			Hint: `
+				dag-scope=entity&entity-bytes=0:x should return a CAR file with only the blocks needed.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", missingBlockFixture.MustGetCidWithCodec(0x70)).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "2200:*"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						HasRoot(missingBlockFixture.MustGetCid()).
+						MightHaveNoRoots().
+						HasBlocks(
+							missingBlockFixture.MustGetCid(),
+							missingBlockFixture.MustGetChildrenCids()[2],
+						).
+						Exactly(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes when missing a block will timeout",
```

Comment:
@laurentsenta (just FYI) I've removed this one, as its body was the same as the first one we had – assuming it was just left-over of some copy-paste.

I did not fix it to actually test timeout, because:
- we don't want to introduce unnecessary latency to the test suite
- the duration after which returning HTTP 504  is not currently specified, so we can't test conformance anyway
- two existing tests are more than enough - especially `entity-bytes=2200:*` one – this alone will fail if implementation is broken the way I described in https://github.com/ipfs/gateway-conformance/issues/75

Reply:
Thanks for the update and removing that useless test 🙏 
I created an issue to add timeouts https://github.com/ipfs/gateway-conformance/issues/89